### PR TITLE
CHANGE: Replaced XR layout builder logic to create Axis2D inputs as Stick controls

### PR DESCRIPTION
**Pull request by jamesmcgill**
### Description
(Migrated fork from https://github.com/Unity-Technologies/InputSystem/pull/1635)

Modifies the XR layout builder logic to add inputs of type Axis2D as Stick controls, instead of Vector2 by default. This enhances functionality for XR controller with additional uses as D-pad, horizontal and vertical axes.

Changes made
XRLayoutBuilder: When a XR Axis2D controller is added to the device layout, replaced layout type "Vector2" with "Stick".

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

**Comment by jamesmcgill** [comment]
@chris-at-unity I think at least some of the failures may be genuine.
I saw XRTests.Layouts_AllFeatureTypes_AreRepresentedInTheGeneratedLayout failing in one of the logs, there might be more.
Could you take a look?

**Comment by chris-at-unity** [comment]
@jamesmcgill sorry on the long delay. I've made the updates on my local branch, but how can I push these to this local branch? It seems I don't have permissions to push them.

**Comment by chris-at-unity** [comment]
nvm, just needed to refresh the SSO credentials associated to my account to push the changes

**Comment by chris-at-unity** [comment]
Re-ran the failed tests and now passed:
https://unity-ci.cds.internal.unity3d.com/job/23355294/results
https://unity-ci.cds.internal.unity3d.com/job/23356316/dependency-graph

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,9 @@ however, it has to be formatted properly to pass verification tests.
 - Fixed issue leading to a stack overflow crash during device initialization in `InsertControlBitRangeNode` (case ISXB-405).
 - Fixed the issue where saving and loading override bindings to JSON would set unassigned overrides (that were `null`) to assigned overrides (as an empty string `""`).
 
+### Changed
+- Changed XR Layout build behavior to create Axis2D control devices with `StickControl` type instead of `Vector2Control`.
```
Comment (jamesmcgill) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: This needs moved up to the Unreleased section. (Probably it was there to begin with, but got left behind after updating the branch).

**Review by jamesmcgill — COMMENTED** [review]
I'll be happy to merge this if you are finished with it?
(after the changelog fix).

File: Packages/com.unity.inputsystem/CHANGELOG.md (hunk 1 above)
Comment (chris-at-unity) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: Done, just pushed the commit with the move

**Comment by chris-at-unity** [comment]
@jamesmcgill sure, you can go forward with the merge, thanks for the help.

**Review by chris-at-unity — APPROVED** [review]

**Comment by jamesmcgill** [comment]
Thanks for PR @chris-at-unity, @RogPodge

**Pull request merged**